### PR TITLE
chore(gen_rpc): Bump version to 3.0.0

### PR DESCRIPTION
**Pull request by ieQu1**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [x] For internal contributor: there is a jira ticket to track this change, and another jira tickt to track doc updates (if any)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

### CODE REVIEW

**Comment by ieQu1** [comment]
Note: I have tested appup with traffic manually in gen_rpc. Additional EMQX cluster update test is needed, does it run in CI?

File: src/emqx.appup.src [hunk 1]
```diff
@@ -6,7 +6,8 @@
      {load_module,emqx_connection,brutal_purge,soft_purge,[]},
      {load_module,emqx_ws_connection,brutal_purge,soft_purge,[]},
      {load_module,emqx_cm,brutal_purge,soft_purge,[]},
-     {load_module,emqx_app,brutal_purge,soft_purge,[]}]},
+     {load_module,emqx_app,brutal_purge,soft_purge,[]},
+     {apply, application, set_env, [gen_rpc, insecure_auth_fallback_allowed, true]}]},
```
Comment (ieQu1) [inline src/emqx.appup.src:10]: Not sure if schema is reloaded during hot upgrade, so I added this upgrade instruction just in case.

**Review by qzhuyan — DISMISSED** [review]

File: changes/v4.4.11-en.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 ### Enhancements
 
+- Update `gen_rpc` dependency to 3.0.0.
```
Comment (thalesmg) [inline changes/v4.4.11-en.md:3]: Small detail:

```suggestion
- Update `gen_rpc` dependency to 3.0.1.
```

**Review by thalesmg — APPROVED** [review]

File: changes/v4.4.11-en.md (hunk 2 above)
Comment (ieQu1) [reply changes/v4.4.11-en.md]: Thanks!

**Review by thalesmg — DISMISSED** [review]

File: changes/v4.4.11-zh.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ### 增强
 
+- 更新`gen_rpc`依赖关系至 3.0.0。
```
Comment (zmstone) [inline changes/v4.4.11-zh.md:3]:
```suggestion
- 更新 `gen_rpc` 依赖关系至 3.0.1。
```

**Comment by zmstone** [comment]
> Note: I have tested appup with traffic manually in gen_rpc standalone application. Additional EMQX cluster update test is needed, does it run in CI?

No.
Please run some manual test before merge.

**Comment by zmstone** [comment]
> > Note: I have tested appup with traffic manually in gen_rpc standalone application. Additional EMQX cluster update test is needed, does it run in CI?
> 
> No. Please run some manual test before merge.

To clarify, we have cluster tests, but not in the github CI

**Review by zmstone — APPROVED** [review]

**Pull request merged**